ui/raidboss/data/05-shb/raid/e12s.js

```diff
@@ -978,6 +978,81 @@ export default {
         },
       },
     },
+    {
+      id: 'E12S Oracle Cataclysm',
+      netRegex: NetRegexes.startsUsing({ source: 'Oracle Of Darkness', id: '58C2', capture: false }),
+      netRegexDe: NetRegexes.startsUsing({ source: 'Orakel Der Dunkelheit', id: '58C2', capture: false }),
+      netRegexFr: NetRegexes.startsUsing({ source: 'Prêtresse Des Ténèbres', id: '58C2', capture: false }),
+      netRegexJa: NetRegexes.startsUsing({ source: '闇の巫女', id: '58C2', capture: false }),
+      delaySeconds: 0.5,
+      promise: async (data, _, output) => {
+        const oracleLocaleNames = {
+          en: 'Oracle Of Darkness',
+          de: 'Orakel Der Dunkelheit',
+          fr: 'Prêtresse Des Ténèbres',
+          ja: '闇の巫女',
+        };
+
+        // select the Oracle Of Darkness
+        let combatantName = null;
+        combatantName = oracleLocaleNames[data.parserLang];
+
+        let combatantData = null;
+        if (combatantName) {
+          combatantData = await window.callOverlayHandler({
+            call: 'getCombatants',
+            names: [combatantName],
+          });
+        }
+
+        // if we could not retrieve combatant data, the
+        // trigger will not work, so just resume promise here
+        if (combatantData === null) {
+          console.error(`Oracle Of Darkness: null data`);
+          data.safeZone = null;
+          return;
+        }
+        if (!combatantData.combatants) {
+          console.error(`Oracle Of Darkness: null combatants`);
+          data.safeZone = null;
+          return;
+        }
+
+        // we need to filter for the Oracle Of Darkness with the highest ID
```

Comment:
Is it the same id as the one who is doing this cast such that you could use `matches.sourceId`?

Reply:
Looks like that's the case to me at least, though I'd need to test it to be sure:
```
20|2021-02-13T22:37:29.2990000-06:00|4001ED5E|Oracle Of Darkness|58C2|Cataclysm|E0000000||5.70||e1c6cf0d8bbd16afba599ca69c645583
```

As such the code above could probably be:
```js
        const combatantData = await window.callOverlayHandler({
          call: 'getCombatants',
          ids: [matches.sourceId],
        });
```

Reply:
I didn't realize it, but yea I can use ids like so to get the correct combatant. Thanks.